iothub/device/src/Transport/RetryDelegatingHandler.cs

```diff
@@ -509,6 +510,12 @@ private Task OpenAsyncInternal(CancellationToken cancellationToken)
 
         private async Task HandleDisconnect()
         {
+            if (_disposed)
+            {
+                if (Logging.IsEnabled) Logging.Info(this, "Race condition: Disposed during disconnection.", nameof(EnsureOpenedAsync));
+                _handleDisconnectCts.Cancel();
```

Comment:
This is leaking the `_transportClosedTask` task.
Regardless if Dispose was called, the task must be observed to avoid a TaskUnobservedException in NET451 and Mono/Xamarin.

Reply:
`_transportClosedTask` is not handled in this class at all. I don't think it will leak a `_transportClosedTask` here.

Reply:
As per above, closing it. Feel free reopen.